server/users.go

```diff
@@ -273,6 +273,21 @@ func (s *Service) UpdateUser(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Don't allow SuperAdmins to modify their own SuperAdmin status.
+	// Allowing them to do so could result in an application where there
+	// are no super admins.
+	ctxUser, ok := hasUserContext(ctx)
+	if !ok {
+		Error(w, http.StatusInternalServerError, "failed to retrieve user from context", s.Logger)
+		return
+	}
+	// If the user being updated is the user making the request and they are
+	// changing their SuperAdmin status, return an unauthorized error
+	if ctxUser.ID == u.ID && req.SuperAdmin != u.SuperAdmin {
```

Comment:
a non-superadmin should never be able to make themselves a superadmin, so i wonder if the more restrictive case (something like `u.SuperAdmin == true && req.SuperAdmin == false`) would be better here to allow other bugs to flow through

Reply:
Doesn't need to be more restrictive, https://github.com/influxdata/chronograf/pull/2639/files/14af1aa1151ebe5ee8455f22b6c9ec6cd6e8597c#diff-9d2bfc0f3a6c230a9c73dbcfd6cdfd53R291

ensures that only super admins can set super admin status

Reply:
I agree it doesn't _need_ to be more restrictive -- my suggestion is to be explicit about the only case that should be viable, so that if we change behavior in the future (and for expectations upon reading), we only catch exactly what is intended there so that other cases continue with execution

Reply:
If you'd like I can change it, this section of code will be changing with 1.4.1 anyways so I don't feel strongly either way.

Reply:
let's change it